docs/resources/client/models/items.md

```diff
@@ -1237,6 +1237,121 @@ protected void registerModels(BlockModelGenerators blockModels, ItemModelGenerat
 </TabItem>
 </Tabs>
 
+## Dynamic Fluid Container
+
+NeoForge adds an item model that constructs a dynamic fluid container, capable of re-texturing itself at runtime to match the contained fluid.
+
+:::note
+For the fluid tint to apply to the fluid texture, the item in question must have a `Capabilities.FluidHandler.ITEM` attached. If your item does not directly use `BucketItem` (not a subtype either), then you need to [register the capability to your item][capability].
+:::
+
+<Tabs>
+<TabItem value="json" label="JSON" default>
+
+```json5
+// For some item 'examplemod:example_item'
+// JSON at 'assets/examplemod/items/example_item.json'
+{
+    "model": {
+        "type": "neoforge:fluid_container",
+
+        // The textures used to construct the container
+        // These are in reference to the block atlas, so they are relative to the `textures` directory
+        "textures": {
+            // Sets the model particle sprite
+            // If not set, uses the first texture that is not null:
+            // - Fluid still texture
+            // - Container base texture
+            // - Container cover texture, if not used as a mask
+            // Points to 'assets/minecraft/textures/item/bucket.png'
+            "particle": "minecraft:item/bucket",
+            // Sets the texture to use on the first layer, generally the container of the fluid
+            // If not set, the layer will not be added
+            // Points to 'assets/minecraft/textures/item/bucket.png'
+            "base": "minecraft:item/bucket",
+            // Sets the texture to use as the mask for the still fluid texture
+            // Areas where the fluid is seen should be pure white
+            // If not set or the fluid is empty, then the layer is not rendered
+            // Points to 'assets/neoforge/textures/item/mask/bucket_fluid.png'
+            "fluid": "neoforge:item/mask/bucket_fluid",
+            // Sets the texture to use as either
+            // - The overlay texture when 'cover_is_mask' is false
+            // - The mask to apply to the base texture (should be pure white to see) when 'cover_is_mask' is true
+            // If not set or no base texture is set when 'cover_is_mask' is true, then the layer is not rendered
+            // Points to 'assets/neoforge/textures/item/mask/bucket_fluid_cover.png'
+            "cover": "neoforge:item/mask/bucket_fluid_cover",
+        },
+
+        // When true, rotates the model 180 degrees
+        // For this to apply, the fluid cannot be empty and its density must be negative or zero
```

Comment:
That's a weird way to phrase it. Remember that the model is dynamic. I would write something like "When true, rotates the model 180 degrees for fluids with negative densities".